### PR TITLE
(TK-394) Pass in domain as keyword only

### PR DESCRIPTION
**Pull request by jpinsonault**
Update tk-metrics to only take domains as keywords

The metrics service originally only accepted domains as strings,
and then eventually it was decided keywords would be better, and
a PR was created to make the registry functions take in either
strings or keywords to maintain backward compatibility until we
decided to make the breaking change.

That moment has arrived, and so this commit removes the last
references to domains as strings.

This PR is part of a larger set of work to move metrics capabilities into OSS puppetserver from PE, see https://tickets.puppetlabs.com/browse/SERVER-1259

### CODE REVIEW

**Comment by camlow325** [comment]
Would it also be possible now to tighten up the schema for [RegistriesConfig](https://github.com/puppetlabs/trapperkeeper-metrics/blob/b0d834f0d6b350f5245b837fdbcd228119517de0/src/clj/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj#L60) to only support `schema/Keyword` for the registry keys instead of `schema/Any`?

File: test/puppetlabs/trapperkeeper/services/metrics/metrics_core_test.clj [hunk 1]
```diff
@@ -13,20 +13,22 @@
 
 (deftest test-initialize-registry-context
   (testing "initializes registry and adds to context"
-    (doseq [domain ["my.epic.domain" :my.epic.domanin]]
-      (let [context (core/initialize-registry-context {} domain)]
-        (is (instance? MetricRegistry (:registry context)))
-        (is (nil? (:jmx-reporter context))))))
+    (let [domain :my.epic.domanin
```
Comment (camlow325) [inline test/puppetlabs/trapperkeeper/services/metrics/metrics_core_test.clj]: nit - guess we could s/domanain/domain while we're in here.  😄

**Comment by camlow325** [comment]
Can we also remove the `Keyword-or-Str` schema from metrics_core.clj since nothing else appears to reference it anymore?

**Comment by jpinsonault** [comment]
Pushed commit addressing the feedback

@rlinehan ping for review when you get a chance

**Comment by rlinehan** [comment]
I think that in `MetricsServiceContext` the `schema/Any`s can be `schema/Keyword`. Otherwise, looks good.

**Comment by jpinsonault** [comment]
Thanks, fixed the schema

File: src/clj/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj [hunk 2]
```diff
@@ -77,7 +77,7 @@
 (def MetricsServiceContext
   {:registries (schema/atom {schema/Any RegistryContext})
```
Comment (rlinehan) [inline src/clj/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj]: I think this can also be `schema/Keyword`

**Comment by jpinsonault** [comment]
D'oh, thanks ruth, amended that commit

**Pull request merged**